test/test_controller.js

```diff
@@ -0,0 +1,364 @@
+            }).then(
+            function (isWhitelisted) {
+                assert.isFalse(isWhitelisted.valueOf(), "1 index address is whitelisted");
+            });
+    });
+});
+
+contract('ICO Controller', function (accounts) {
+    it("test addAirdrop", function () {
+        var addAccounts = [accounts[1], accounts[2]];
+        var controller_instance;
+        return Controller.deployed().then(
+            function (inst) {
+                controller_instance = inst;
+                return controller_instance.airdropList.call(addAccounts[0]);
+            }).then(
+            function (inDroplist) {
+                assert.isFalse(inDroplist.valueOf(), "account in airdrop list by default");
+                return controller_instance.totalAirdropAdrresses.call();
+            }).then(
+            function (totalAdrresses) {
+                assert.equal(totalAdrresses.valueOf(), 0, "account total number is not 0 by default");
+                return controller_instance.addAirdrop(addAccounts);
+            }).then(
+            function () {
+                return controller_instance.totalAirdropAdrresses.call();
+            }).then(
+            function (total_addr) {
+                assert.equal(total_addr.valueOf(), addAccounts.length, "Expect totalAirdropAdrresses become: " + addAccounts.length);
+                return controller_instance.airdropList.call(addAccounts[0]);
+            }).then(
+            function (inDroplist) {
+                assert.isTrue(inDroplist.valueOf(), "Excpect Account 0 to be in airdrop list");
+                return controller_instance.airdropList.call(addAccounts[1]);
+            }).then(
+            function (inDroplist) {
+                assert.isTrue(inDroplist.valueOf(), "Excpect Account 1 to be in airdrop list");
+            });
+    });
+
+    it("test removeAirdrop", function () {
+        var addAccounts = [accounts[3], accounts[4]];
+        var controller_instance;
+        return Controller.deployed().then(
+            function (inst) {
+                controller_instance = inst;
+                return controller_instance.addAirdrop(addAccounts);
+            }).then(
+            function () {
+                return controller_instance.removeAirdrop(addAccounts);
+            }).then(
+            function () {
+                return controller_instance.airdropList.call(addAccounts[0]);
+            }).then(
+            function (inDroplist) {
+                assert.isFalse(inDroplist.valueOf(), "Excpect Account 0 not to be in airdrop list");
+                return controller_instance.airdropList.call(addAccounts[1]);
+            }).then(
+            function (inDroplist) {
+                assert.isFalse(inDroplist.valueOf(), "Excpect Account 1 not to be in airdrop list");
+            });
+    });
+});
+
+contract('ICO Controller', function (accounts) {
+    it("test getAirdropTokens", function () {
```

Comment:
I think that it is worth testing the case when the number of tokens is not divided by the number of recipients